### ISSUE TITLE
sm-radio-button design questions

### ISSUE THREAD

**Issue by m0t0r** [Semantic-Org/Semantic-UI-Angular#5]
I am planning to create a radio button directive (sm-radio-button) which should probably work in conjunction with sm-radio-group directive which manages a set of  sm-radio-button directive. This design is borrowed from angular-material. So, in generic case it should look like:
```js
<sm-radio-group ng-model="radio">
    <sm-radio-button value="radio1">Radio 1</sm-radio-button>
    <sm-radio-button value="radio2">Radio 2</sm-radio-button>
    <sm-radio-button value="radio3">Radio 3</sm-radio-button>
</sm-radio-group>
```

I have a couple of questions:

1) Angular-design has a set of custom services such as '$mdConstant', '$mdUtil'. Should I consider analogues services as well since they look like good ideas.

2) Is design using controller's prototype to set helper methods considered suitable for us as well ?

3) What else should I consider ?

@caitp please take a look once you have a chance.

**Comment by HackedByChinese** [comment]
If the outcome of this discussion is to set precedent for organizing modules and how the modules are named, I'd like to throw in my $.02.

1) I agree that functional components (either supporting services used by multiple directives, or individual directives themselves) should be in separate modules beneath the main module itself. This is not so much, I don't think, because we want to concern ourselves with letting people pick and choose which modules they need (although it's not a bad idea; the breadth of Semantic UI is large and not all of it is relevant to each project), but to isolate functionality for testing and bring in dependencies as needed. Having written a few modules and some large apps myself with the starting thought of "why segment; I'll just throw it all in one place to save time and effort", I've almost always regretted it as you maintain the project over time.

2) Brevity and simple but effective organization is key. Regarding the names, I would avoid redundancy, unnecessary explicitness, and keep things canonical. Using ui-bootstrap as a baseline, I prefer this pattern: `ui.semantic.[functional component name]`. No need to add in unnecessary stuff like `.angular` (it's an angular module; we already know it's angular, and there's no significant way to mix angular with non angular code to the extent we have to worry about collisions) or `.elements` (not really a standard in the lexicon of Angular modules, and I don't think there's much value in over-categorizing, plus everything in this module is either UI or supporting UI). Examples: `ui.semantic.button`, `ui.semantic.modal`, `ui.semantic.position` (a theoretical shared component that supports calculating positions for tooltips and the likes), so on and so forth.

**Comment by Darkeye9** [comment]
@HackedByChinese I think your comment should be put under #8 (About this project).
I definitely agree that we need to set some guidelines and then write some example modules to test them. Once those guidelines are established, we can start the coding race ;)

**Comment by m0t0r** [comment]
Closed by #10
